src/core/util/formats.cpp

```diff
@@ -67,285 +69,118 @@ size_t commonFormatPrecision(std::span<const DataFormatBase*> formats) {
-                        case 8:
-                            return DataVec4Int8::get();
-                        case 16:
-                            return DataVec4Int16::get();
-                        case 32:
-                            return DataVec4Int32::get();
-                        case 64:
-                            return DataVec4Int64::get();
-                    }
-                    break;
-            }
-            break;
-        case NumericType::UnsignedInteger:
-            switch (components) {
-                case 1:
-                    switch (precision) {
-                        case 8:
-                            return DataUInt8::get();
-                        case 16:
-                            return DataUInt16::get();
-                        case 32:
-                            return DataUInt32::get();
-                        case 64:
-                            return DataUInt64::get();
-                    }
-                    break;
-                case 2:
-                    switch (precision) {
-                        case 8:
-                            return DataVec2UInt8::get();
-                        case 16:
-                            return DataVec2UInt16::get();
-                        case 32:
-                            return DataVec2UInt32::get();
-                        case 64:
-                            return DataVec2UInt64::get();
-                    }
-                    break;
-                case 3:
-                    switch (precision) {
-                        case 8:
-                            return DataVec3UInt8::get();
-                        case 16:
-                            return DataVec3UInt16::get();
-                        case 32:
-                            return DataVec3UInt32::get();
-                        case 64:
-                            return DataVec3UInt64::get();
-                    }
-                    break;
-                case 4:
-                    switch (precision) {
-                        case 8:
-                            return DataVec4UInt8::get();
-                        case 16:
-                            return DataVec4UInt16::get();
-                        case 32:
-                            return DataVec4UInt32::get();
-                        case 64:
-                            return DataVec4UInt64::get();
-                    }
-                    break;
-            }
-            break;
-        case NumericType::NotSpecialized:
-        default:
-            break;
+    static const auto table = []() {
+        std::array<std::array<std::array<const DataFormatBase*, 2>, 4>, 4> res = {nullptr};
+        util::for_each_type<DefaultDataFormats>{}([&]<typename F>() {
+            const auto n = static_cast<size_t>(F::numericType());
+            const auto c = F::components();
+            const auto p = F::precision() / 8 - 1;
+            res[n][c][p] = F::get();
```

Comment:
This might need looking into.

> Check warning on line 170 in src/core/util/formats.cpp
> GitHub Actions / Build (linux, dynamic, Release)
> array subscript 33 is outside array bounds of ‘const std::array<std::array<std::array<const inviwo::DataFormatBase*, 2>, 4>, 4>